### ISSUE TITLE
Move _SETUP.md content to CONTRIBUTING.md

### ISSUE THREAD

**Issue by jeremenichelli** [webpack/webpack#6893]
**Do you want to request a *feature* or report a *bug*?**

Feature (?)

**What is the current behavior?**

The list of commands needed to run before developing are in `_SETUP.md` content.

**If the current behavior is a bug, please provide the steps to reproduce.**

In the `CONTRIBUTING.md` file the setup link is easy to miss.

**What is the expected behavior?**

SETUP list of commands are easier to find.

**If this is a feature request, what is motivation or use case for changing the behavior?**

Since the content of the SETUP file is really not big I think we should move them to CONTRIBUTING so developer can discover easily the steps needed before running tests.

**Please mention other relevant information such as the browser version, Node.js version, webpack version, and Operating System.**

--

**Comment by ooflorent** [comment]
@Legends: Could you share your thoughts on this?

**Comment by montogeek** [comment]
Yes, everything should be in one file, no need to create more files to read, if people don't read 1 file less 2 files.

**Comment by Legends** [comment]
I would initially agree with @montogeek, a separate file for ~5 lines of information is a bit of overhead.

**Let me explain it:**
I have extracted the setup information and added an `_` to the filename to make it appear first of all files in a folder.
The thing is that other files like `Contributing.md`,`Readme.md` don't currently have this underscore and they will appear somewhere in the middle of all `.js` files.

If all `*.md` files had a leading underscore they would be grouped together, then you cannot miss the `_SETUP.md` file.

Like seen in the picture below (folder test), the instructions are somewhere in the middle of all files but should come first. People who start contributing to webpack, will see them **instantly** if they all had an underscore and they would be **grouped together**, if there are multiple. The convention should be all `*.md` files should start with an `_`.

We have approximately 95 `.md` files in webpack.
Instructions (*.md) should come first, imo.
And the first thing when you start a new project you are looking for the `setup` instructions.

As always, the majority rules over the minority ;-) So, if people don't agree, I can revert it back, no problem.

![image](https://user-images.githubusercontent.com/966368/38044258-8732625e-32b9-11e8-80fe-fd2f110e4d61.png)

**Comment by Legends** [comment]
Btw. there is an [open PR for bootstrapping the setup](https://github.com/webpack/webpack/pull/6880). 

So the setup instructions will finally be a one-liner and not a list of commands anymore:

`npm run setup`  --> to explicitly call setup

**Actually** you will not need to do anything for setting up the project, just write your customizations, your tests and run them:
`npm run test`  --> will call setup implicitly and will skip the setup if the project is already setup.

**Comment by Legends** [comment]
After the PR above is merged, we could indeed delete the `_SETUP.md`.
We can put it back to `Contributing.md` and add some information about the new setup way.

Because no one will bother with `how to setup the local webpack repo` anymore.

**Comment by Legends** [comment]
I will submit a PR for this

**Comment by Janpot** [comment]
Maybe only mildly relevant, but I made my first PR this morning. Finding the setup instructions was not hard, but skipping a step and combining the files will slightly improve the experience. Same for combining the commands into one, you just have to do this once anyway and copy+pasting a block of 3 commands, or a single command isn't that much different. Imo, these are just minor improvements.

But, my number one pain point in contributing to webpack was finding ways to selectively run test cases. Running the full suite takes several minutes. I like to use a test driven approach when debugging and I had to reverse engineer the test framework and patch it with one-off code to get single test cases to run. If there's a way to simplify this, that would make contributing a whole lot easier.

**Comment by Legends** [comment]
Of course it's only a minor improvement, but the issue pops up under issues, from time to time.
Like me for example, I just wanted to test something out and it failed, now it will complete.

You mean selectively executing **existing** tests?  Open a PR for this.

**Comment by Janpot** [comment]
sry, didn't mean to diminish the effort in this issue.

**Comment by Legends** [comment]
Everything is fine ;-) I get your point. Like I said, open an issue for `how to selectively run test cases`.

**Comment by Janpot** [comment]
Sure, I have some uncommited leftovers that does this. Will see if I can generalize it in a PR later.

**Comment by jeremenichelli** [comment]
Yeap, I recently submitted a PR and performance improvements and run selectively test cases will be both really welcomed.

**Comment by montogeek** [comment]
@Janpot That would be fixed once tests are migrated to Jest #6565

**Comment by Legends** [comment]
I have merged `_SETUP.md` back to `CONTRIBUTING.md` [here](https://github.com/Legends/webpack/blob/webpack-bootstrap-setup/CONTRIBUTING.md#setup), it will be available when the PR gets merged.

**Comment by Legends** [comment]
I guess, the _SETUP.md will nevertheless be a separate md file. Close?

**Comment by jeremenichelli** [comment]
@Legends I think that though we've added a setup command, the content is minimal and itsmore suitable to be into _contirbuting_. Three deep linked pages for such a minimal content makes no sense in my opinion.

**Comment by Legends** [comment]
I understand you and it's not on me to decide.
It was separated to increase its visibility, in order to be found easier by new contributors.
This should also be the only use-case where we split out information from an existing `.md` into a separate one, to prevent bloating the repo.

Another thing I would do, is that every `md` files starts with an underscore, in order to group them together and let them appear first after all folders.... my 2c

**Comment by Legends** [comment]
I have merged it back in my PR, sokra made a new one, based on mine... and he kept it as a separate `md`...
Just get in touch with him... if he is alright with that, we can reverse it easily..
